index.ts

```diff
@@ -964,6 +964,38 @@ export class JSONValue {
     return this.kind == JSONValueKind.NULL
   }
 
+  isBool(): boolean {
+    return this.kind == JSONValueKind.BOOL
+  }
+
+  isI64(): boolean {
+    return this.kind == JSONValueKind.NUMBER
+  }
+
+  isU64(): boolean {
+    return this.kind == JSONValueKind.NUMBER
+  }
+
+  isF64(): boolean {
+    return this.kind == JSONValueKind.NUMBER
+  }
```

Comment:
These are still a little ambiguous, which is unavoidable if we only check `== JSONValueKind.NUMBER`. Making these accurate would require `json.isI64()`, `json.isU64()` and `json.isF64()` host exports that basically do e.g. `i64::from_str(json_str).is_ok()`...

If we want to be correct, then that's what we should do.